File: foodstore-store/src/lib/types/order.ts
export interface Order {
	id: string;
	orderCode: string;
	branchId?: string;
	branchName?: string;
	kitchenRouting?: string;
	sourceId?: string;
	sourceName?: string;
	source?: { id: number; name: string };
	customerId?: number;
	items: OrderItem[];
	subtotalAmount: number;
	discountId?: number;
	discountAmount: number;
	discountCode?: string;
	vatAmount?: number;
	totalAmount: number;
	status: string;

	note?: string;
	customerName?: string;
	customerPhone?: string;
	history?: OrderStatusHistory[];
	qrPaymentUrl?: string;
	createdAt: string;
	updatedAt?: string;
}

export interface OrderItem {
	id?: number;
	menuItemId: number;
	menuItemName?: string;
	comboId?: number;
	comboName?: string;
	quantity: number;
	unitPrice: number;
	totalPrice: number;
	addons?: OrderAddon[];
	note?: string;
}

export interface OrderAddon {
	addonId: number;
	addonName?: string;
	quantity: number;
	// Backend sends unitPrice, not price.
	unitPrice: number;
}

export type OrderStatus =
	| 'pending'
	| 'confirmed'
	| 'preparing'
	| 'ready'
	| 'served'
	| 'paid'
	| 'cancelled';

export interface OrderCreateDto {
	sourceId?: string;
	customerName?: string;
	customerPhone?: string;
	items: OrderItemCreateDto[];
	discountId?: number;
	discountCode?: string;
	note?: string;
}

export interface OrderItemCreateDto {
	menuItemId?: number;
	comboId?: number;
	quantity: number;
	addons?: { addonId: number; quantity: number }[];
	note?: string;
}

export interface OrderUpdateStatusDto {
	status: string;
	paymentMethod?: string;
	paymentAmount?: number;
}

export interface OrderStatusHistory {
	id: number;
	orderId: number;
	fromStatus: string;
	toStatus: string;
	note?: string;
	changedBy: number;
	changedAt: string;
	changedByName?: string;
}

File: foodstore-api/FoodstoreApi.Core/Entities/Order.cs
namespace FoodstoreApi.Core.Entities;

public partial class Order : IAuditableEntity
{
    public Guid Id { get; set; }
    public string OrderCode { get; set; } = null!;
    public Guid? SourceId { get; set; }
    public Guid? BranchId { get; set; }
    public Guid? EmployeeId { get; set; }
    public Guid? CustomerId { get; set; }
    public Guid? DiscountId { get; set; }
    public decimal? SubtotalAmount { get; set; }
    public decimal? DiscountAmount { get; set; }
    public decimal? VatAmount { get; set; }
    public decimal? TotalAmount { get; set; }
    public string? QrPaymentUrl { get; set; }
    public DateTime? PaidAt { get; set; }
    public string? Status { get; set; }
    public string? Note { get; set; }
    public DateTime? PrintedAt { get; set; }
    public Guid? UpdatedBy { get; set; }
    public Guid? CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public virtual Discount? Discount { get; set; }
    public virtual Customer? Customer { get; set; }
    public virtual Employee? Employee { get; set; }
    public virtual Source? Source { get; set; }
    public virtual Branch? Branch { get; set; }
    public virtual Employee? UpdatedByNavigation { get; set; }
    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public virtual ICollection<OrderStatusHistory> OrderStatusHistories { get; set; } = new List<OrderStatusHistory>();
    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();
}
